src/changelog.rs

```diff
@@ -0,0 +1,64 @@
+use crate::*;
+use std::collections::VecDeque;
+
+#[derive(Clone)]
+#[near(serializers=[borsh, json])]
+pub struct ChangeLog {
+    pub block_id: u64,
+    pub block_timestamp: u64,
+    pub change_log_type: ChangeLogType,
+}
+
+#[derive(Clone)]
+#[near(serializers=[borsh, json])]
+pub enum ChangeLogType {
+    Proposal(ProposalId),
+    RFP(RFPId),
+}
+
+// The changelog will be a FIFO queue of changes with the length of 50.
+impl Contract {
+    pub fn add_change_log(&mut self, change_log_type: ChangeLogType) {
+        let new_log = ChangeLog {
+            block_id: env::block_height(),
+            block_timestamp: env::block_timestamp(),
+            change_log_type,
+        };
+        if self.change_log.len() >= 50 {
+            self.change_log.pop_front();
+        }
+        self.change_log.push_back(new_log);
+    }
+}
+
+// TODO remove this if not necessary
+#[near(serializers=[borsh, json])]
+pub struct ChangeLogQueue(pub VecDeque<ChangeLog>);
+
+// TODO remove this
+// Add methods to delegate to the inner VecDeque
+impl ChangeLogQueue {
```

Comment:
I feel it is a pointless abstraction since `Contract::add_change_log` handles the "queue" logic, so I would just remove this implementation as useless.